### PR TITLE
Use port z0 for plot_z_time_step, plot_z_time_impulse, restore missing ylabel and add modes subscript to plot trace names for mixed-modes networks

### PR DESCRIPTION
**Pull request by mhuser**
- Make both `plot_z_time_step` and `plot_z_time_step` use the reference impedances of each port. For mixed-modes networks or other specific cases, all ports' reference impedances are not identical.
- Subscripts are added to trace names for all attributes if the `port_modes` are not both single-ended, like `Sdd11, Scd13`.
- Fix missing plot ylabel that appeared in 1.1.0
- Add test cases and update some notebooks to reflect the changes.

348473686-51a79422-63ea-490e-9455-d2de05001b30

348473698-1f32b649-bea9-4b1a-b0f9-a25bac011c5a

348491285-b3139298-07ef-4ec9-8532-5a527d467f6a

### CODE REVIEW

**Comment by mhuser** [comment]
With #1124 
This should ease plotting syntax a lot for mixed-modes !

image

image

image

```Python
import skrf as rf
from skrf.media import DefinedGammaZ0
from scipy.constants import  c, pi
import matplotlib.pyplot as plt
rf.stylely()

f = rf.Frequency(0.01, 2, 200, 'GHz')
beta  = 2 * pi * f.f / c # propagation in air
m = DefinedGammaZ0(frequency = f, z0_port = 50, gamma = 1j * beta)
l = m.line(150e-3, 'm', z0 = 55, name = 'line')
diff = rf.concat_ports([l, l], port_order='second')
diff.name = 'diff line'
diff.se2gmm(p = 2)
l = l.extrapolate_to_dc(kind = 'cubic')
diff = diff.extrapolate_to_dc(kind = 'cubic')

print(diff.z0[0])

figure = plt.figure()
plt.title('Step response')
l.plot_z_time_step(0, 0)
plt.xlim((-1.5, 2.5))

figure = plt.figure()
plt.title('Step response')
diff.plot_z_time_step(0, 0)
diff.plot_z_time_step(2, 2)
diff.plot_z_time_step(2, 0)
diff.plot_z_time_step(0, 2)
plt.xlim((-1.5, 2.5))

figure, axs = plt.subplots(1, 2, sharex = True)
plt.suptitle('Impulse response')
diff.plot_z_time_impulse(0, 0, ax = axs[0])
diff.plot_z_time_impulse(2, 2, ax = axs[1])
diff.plot_z_time_impulse(2, 0, ax = axs[0])
diff.plot_z_time_impulse(0, 2, ax = axs[1])
plt.xlim((-1.5, 2.5))
plt.tight_layout()
```

**Comment by mhuser** [comment]
image

Gorgeous, but it has some side effects

**Comment by mhuser** [comment]
Depending how the network is created (ex. init first and later s set), `port_modes` are a not properly filled. This was not a problem as long as we did not use them.

**Comment by mhuser** [comment]
This syntax results in `port_modes` of length 0. `Network` is initialized without ports or s parameters and later `s` is set.

https://github.com/scikit-rf/scikit-rf/blob/b67a5c2dacaf7415991c2606716ade9d5a140eab/skrf/calibration/calibration.py#L6683-L6702

**Comment by mhuser** [comment]
https://scikit-rf--1124.org.readthedocs.build/en/1124/examples/mixedmodeanalysis/Mixed%20Mode%20Basics.html

image

This PR should be review-ready.

**Comment by mhuser** [comment]
seems `Network.Y_LABEL_DICT` is not used by plot anymore

File: skrf/network.py [hunk 1]
```diff
@@ -4692,7 +4720,9 @@ def plot_attribute(     self,
                             y_label=None,
                             logx=False, **kwargs):
 
-
+        # create y_label if not provided
+        if y_label is None:
+            y_label = Network.Y_LABEL_DICT[conversion]
```
Comment (mhuser) [inline skrf/network.py:4725]: @FranzForstmayr I am not sure it is the most proper way to make plots have y_label again. Have you eventually other ideas with the autogenerated plot functions? (cf. #1125)

Comment (mhuser) [reply skrf/network.py:4725]: At least, ylabel are back!
image

File: skrf/calibration/calibration.py [hunk 2]
```diff
@@ -5232,7 +5232,7 @@ def __init__(self, measured, ideals, ideal_is_reflect=True, sign=None,
         if switch_terms is None:
             warn('No switch terms provided', stacklevel=2)
 
-        if type(ideals) == Network:
+        if type(ideals) is Network:
```
Comment (FranzForstmayr) [inline skrf/calibration/calibration.py:5235]:
```suggestion
        if isinstance(ideals, Network):
```

Comment (mhuser) [reply skrf/calibration/calibration.py]: Better! Done.

**Comment by mhuser** [comment]
@FranzForstmayr 
Do you think I dare merge this one?

**Pull request merged**